File: backend/src/main/java/com/jorge/gymprogress/model/RoutineExercise.java
package com.jorge.gymprogress.model;

import jakarta.persistence.*;

@Entity
@Table(name = "routine_exercises")
public class RoutineExercise {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // Rutina a la que pertenece este ejercicio
    @ManyToOne(optional = false)
    @JoinColumn(name = "routine_id", nullable = false)
    private Routine routine;

    // Ejercicio que se añade a la rutina
    @ManyToOne(optional = false)
    @JoinColumn(name = "exercise_id", nullable = false)
    private Exercise exercise;

    // Orden en el que aparece el ejercicio dentro de la rutina
    @Column(name = "exercise_order")
    private Integer exerciseOrder;

    @Column(nullable = false)
    private Integer targetSets;

    @Column(nullable = false)
    private String targetRepetitions;

    private Integer restSeconds;

    private String notes;

    public RoutineExercise() {
    }

    public Long getId() {
        return id;
    }

    public Routine getRoutine() {
        return routine;
    }

    public Exercise getExercise() {
        return exercise;
    }

    public Integer getExerciseOrder() {
        return exerciseOrder;
    }

    public Integer getTargetSets() {
        return targetSets;
    }

    public String getTargetRepetitions() {
        return targetRepetitions;
    }

    public Integer getRestSeconds() {
        return restSeconds;
    }

    public String getNotes() {
        return notes;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public void setRoutine(Routine routine) {
        this.routine = routine;
    }

    public void setExercise(Exercise exercise) {
        this.exercise = exercise;
    }

    public void setExerciseOrder(Integer exerciseOrder) {
        this.exerciseOrder = exerciseOrder;
    }

    public void setTargetSets(Integer targetSets) {
        this.targetSets = targetSets;
    }

    public void setTargetRepetitions(String targetRepetitions) {
        this.targetRepetitions = targetRepetitions;
    }

    public void setRestSeconds(Integer restSeconds) {
        this.restSeconds = restSeconds;
    }

    public void setNotes(String notes) {
        this.notes = notes;
    }
}

File: frontend/lib/models/routine_exercise.dart
class RoutineExercise {
  final int id;

  final int routineId;
  final String routineName;

  final int exerciseId;
  final String exerciseName;
  final String muscleGroup;

  final int exerciseOrder;
  final int targetSets;
  final String targetRepetitions;
  final int? restSeconds;
  final String? notes;

  RoutineExercise({
    required this.id,
    required this.routineId,
    required this.routineName,
    required this.exerciseId,
    required this.exerciseName,
    required this.muscleGroup,
    required this.exerciseOrder,
    required this.targetSets,
    required this.targetRepetitions,
    required this.restSeconds,
    required this.notes,
  });

  factory RoutineExercise.fromJson(Map<String, dynamic> json) {
    return RoutineExercise(
      id: json['id'],
      routineId: json['routineId'],
      routineName: json['routineName'] ?? '',
      exerciseId: json['exerciseId'],
      exerciseName: json['exerciseName'] ?? '',
      muscleGroup: json['muscleGroup'] ?? '',
      exerciseOrder: json['exerciseOrder'] ?? 0,
      targetSets: json['targetSets'] ?? 0,
      targetRepetitions: json['targetRepetitions'] ?? '',
      restSeconds: json['restSeconds'],
      notes: json['notes'],
    );
  }
}